app/models/docker_registry.rb

```diff
@@ -0,0 +1,42 @@
+class DockerRegistry < ActiveRecord::Base
+  belongs_to :certificate, polymorphic: true
+
+  validates :name, presence: true
+
+  class << self
+    def apply(settings_params)
+      settings_params["insecure_registries"].each do |registry|
+        next if registry["docker_registry_name"].blank?
+        certificate = unless registry["docker_registry_certificate"].blank?
+          Certificate.find_or_initialize_by(
+            certificate: registry["docker_registry_certificate"]
+          ).tap do |c|
+            c.certificate = registry["docker_registry_certificate"]
+            c.save
+          end
+        end
+        self.find_or_initialize_by(
+          name: registry["docker_registry_name"]
+        ).tap do |r|
+          r.name = registry["docker_registry_name"]
+          r.certifiable_id = certificate.blank? ? nil : certificate.id
```

Comment:
I would use `nil?` instead of `blank?` since we are expecting `nil` or an object from the `unless` statement.